backend/models/shapes.js

```diff
@@ -0,0 +1,73 @@
+/**
+ * Copyright (C) 2021 3D Repo Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+"use strict";
+const DB = require("../handler/db");
+const yup = require("yup");
+
+const utils = require("../utils");
+const responseCodes = require("../response_codes.js");
+const getCollectionName = (model, subCollectionName) => `${model}.${subCollectionName}.shapes`;
+
+const coordinatesSchema = yup.array().of(yup.number()).length(3);
+const colorSchema = yup.array().of(yup.number().min(0).max(1)).length(4);
+
+const shapesSchema = yup.object().shape({
+	"positions": yup.array().of(coordinatesSchema).required(),
+	"normals": yup.array().of(coordinatesSchema),
+	"value": yup.number().min(0),
+	"color": colorSchema.required(),
+	"type": yup.mixed().oneOf([0, 1]).required(),
+	"ticket_id": utils.uuidSchema,
+	"name": yup.string(),
+	// this is for convenience only, because the shape has this field in unity we allow it
+	// to be posted in this form, even though it will be replaced with _id when fetching
+	"uuid": yup.string()
+}).noUnknown();
+
+const Shapes = {};
+
+Shapes.clean = (shape) => {
+	shape.uuid = utils.uuidToString(shape._id);
```

Comment:
I think some sort of Id, might be necessary in the future to identifying each shape. Right now is not so important because the whole shapes collection are being updated so its only necessary to have an uuid (or _id) in the frontend. However if we want to maybe improve the performance in the future of that and have special messages for realtime chat events, ids that are consistent across the clients are necessary.
My question is:
Should I get rid of this for now or should I rename it to _id  for consistency across the backend?

Reply:
I think if we want an ID for the shapes, we should just utilise _id.

Otherwise it's just confusing to anyone who wants to use the APIs and wonder why do we have a uuid and an _id 😆 

both of them serves the purpose of identifying a shape anyway. and the uuid logic is really just something between the front end and the unity viewer